--- glib-2.36.0/glib/Makefile.am.makefile	2013-03-25 22:56:57.103911399 +0100
+++ glib-2.36.0/glib/Makefile.am	2013-03-25 22:57:04.707203541 +0100
@@ -326,11 +326,6 @@ CLEANFILES += glib-public-headers.txt
 all-local: glib-public-headers.txt
 
 install-data-local: install-ms-lib
-	@if test -f $(glibincludedir)/glist.h ; then					\
-	  echo "*** Old headers found in $(glibincludedir). You should remove the" ;	\
-	  echo "*** contents of this directory and type 'make install' again." ;	\
-	  false ;									\
-	fi
 
 uninstall-local: uninstall-ms-lib uninstall-gdb
 
