diff -urN glib-2.1.3.orig/glib/Makefile.am glib-2.1.3/glib/Makefile.am
--- glib-2.1.3.orig/glib/Makefile.am	Sat Nov  9 12:33:56 2002
+++ glib-2.1.3/glib/Makefile.am	Sat Nov  9 12:53:24 2002
@@ -143,8 +143,8 @@
 	gwin32.h
 
 install-data-local: install-ms-lib install-libtool-import-lib
-	@if test -f $(glibincludedir)/glist.h ; then					\
-	  echo "*** Old headers found in $(glibincludedir). You should remove the" ;	\
+	@if test -f $(DESTDIR)$(glibincludedir)/glist.h ; then					\
+	  echo "*** Old headers found in $(DESTDIR)$(glibincludedir). You should remove the" ;	\
 	  echo "*** contents of this directory and type 'make install' again." ;	\
 	  false ;									\
         fi
