diff -Nur glib-2.12.4.orig/configure.in glib-2.12.4.pkgconfig-fix/configure.in
--- glib-2.12.4.orig/configure.in	2006-10-01 05:51:15.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/configure.in	2006-11-05 20:44:19.000000000 +0000
@@ -863,8 +863,10 @@
   AC_CHECK_LIB(rt, clock_gettime, [
     AC_DEFINE(HAVE_CLOCK_GETTIME, 1)
     LIBS="$LIBS -lrt"
+    LIBRT="-lrt"
   ])
 ])
+AC_SUBST(LIBRT)
 
 AC_CACHE_CHECK(for monotonic clocks,
     glib_cv_monotonic_clock,AC_TRY_RUN([
diff -Nur glib-2.12.4.orig/glib-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/glib-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/glib-2.0-uninstalled.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/glib-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -5,5 +5,5 @@
 Name: GLib Uninstalled
 Description: C Utility Library, Not Installed
 Version: @VERSION@
-Libs: ${pc_top_builddir}/${pcfiledir}/glib/libglib-2.0.la @ICONV_LIBS@ @INTLLIBS@
+Libs: ${pc_top_builddir}/${pcfiledir}/glib/libglib-2.0.la @ICONV_LIBS@ @INTLLIBS@ @LIBRT@
 Cflags: -I${pc_top_builddir}/${pcfiledir} -I${pc_top_builddir}/${pcfiledir}/@srcdir@ -I${pc_top_builddir}/${pcfiledir}/@srcdir@/glib
diff -Nur glib-2.12.4.orig/glib-2.0.pc.in glib-2.12.4.pkgconfig-fix/glib-2.0.pc.in
--- glib-2.12.4.orig/glib-2.0.pc.in	2003-07-30 00:01:39.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/glib-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -10,6 +10,6 @@
 Name: GLib
 Description: C Utility Library
 Version: @VERSION@
-Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@
+Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@ @LIBRT@
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include 
 
diff -Nur glib-2.12.4.orig/gmodule-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/gmodule-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/gmodule-2.0-uninstalled.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gmodule-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -4,5 +4,5 @@
 Description: Dynamic module loader for GLib, Not Installed
 Requires: glib-2.0-uninstalled
 Version: @VERSION@
-Libs: @G_MODULE_LDFLAGS@ ${pc_top_builddir}/${pcfiledir}/gmodule/libgmodule-2.0.la @G_MODULE_LIBS@
+Libs: @G_MODULE_LDFLAGS@ ${pc_top_builddir}/${pcfiledir}/gmodule/libgmodule-2.0.la @G_MODULE_LIBS@ @LIBRT@
 Cflags: -I${pc_top_builddir}/${pcfiledir}/@srcdir@/gmodule
diff -Nur glib-2.12.4.orig/gmodule-2.0.pc.in glib-2.12.4.pkgconfig-fix/gmodule-2.0.pc.in
--- glib-2.12.4.orig/gmodule-2.0.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gmodule-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -9,5 +9,5 @@
 Description: Dynamic module loader for GLib
 Requires: glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} @G_MODULE_LDFLAGS@ -lgmodule-2.0 @G_MODULE_LIBS@
+Libs: -L${libdir} @G_MODULE_LDFLAGS@ -lgmodule-2.0 @G_MODULE_LIBS@ @LIBRT@
 Cflags:
diff -Nur glib-2.12.4.orig/gmodule-export-2.0.pc.in glib-2.12.4.pkgconfig-fix/gmodule-export-2.0.pc.in
--- glib-2.12.4.orig/gmodule-export-2.0.pc.in	2005-01-07 15:20:15.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gmodule-export-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -9,5 +9,5 @@
 Description: Dynamic module loader for GLib
 Requires: glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} @G_MODULE_LDFLAGS@ -lgmodule-2.0 @G_MODULE_LIBS@
+Libs: -L${libdir} @G_MODULE_LDFLAGS@ -lgmodule-2.0 @G_MODULE_LIBS@ @LIBRT@
 Cflags:
diff -Nur glib-2.12.4.orig/gmodule-no-export-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/gmodule-no-export-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/gmodule-no-export-2.0-uninstalled.pc.in	2005-05-06 21:54:52.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gmodule-no-export-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -9,6 +9,6 @@
 Description: Dynamic module loader for GLib
 Requires: glib-2.0
 Version: @VERSION@
-Libs: ${pc_top_builddir}/${pcfiledir}/gmodule/libgmodule-2.0.la
+Libs: ${pc_top_builddir}/${pcfiledir}/gmodule/libgmodule-2.0.la @LIBRT@
 Cflags: -I${pc_top_builddir}/${pcfiledir}/@srcdir@/gmodule
 
diff -Nur glib-2.12.4.orig/gmodule-no-export-2.0.pc.in glib-2.12.4.pkgconfig-fix/gmodule-no-export-2.0.pc.in
--- glib-2.12.4.orig/gmodule-no-export-2.0.pc.in	2004-11-17 03:33:05.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gmodule-no-export-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -9,5 +9,5 @@
 Description: Dynamic module loader for GLib
 Requires: glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} -lgmodule-2.0 @G_MODULE_LIBS@
+Libs: -L${libdir} -lgmodule-2.0 @G_MODULE_LIBS@ @LIBRT@
 Cflags:
diff -Nur glib-2.12.4.orig/gobject-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/gobject-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/gobject-2.0-uninstalled.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gobject-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -2,6 +2,6 @@
 Description: Object/type system for GLib, Not Installed
 Requires: glib-2.0-uninstalled
 Version: @VERSION@
-Libs: ${pc_top_builddir}/${pcfiledir}/gobject/libgobject-2.0.la
+Libs: ${pc_top_builddir}/${pcfiledir}/gobject/libgobject-2.0.la @LIBRT@
 ## cflags contains builddir in addition to srcdir because of gmarshal.h
 Cflags: -I${pc_top_builddir}/${pcfiledir}/@srcdir@ -I${pc_top_builddir}/${pcfiledir}
diff -Nur glib-2.12.4.orig/gobject-2.0.pc.in glib-2.12.4.pkgconfig-fix/gobject-2.0.pc.in
--- glib-2.12.4.orig/gobject-2.0.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gobject-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -7,5 +7,5 @@
 Description: GLib Type, Object, Parameter and Signal Library
 Requires: glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} -lgobject-2.0
+Libs: -L${libdir} -lgobject-2.0 @LIBRT@
 Cflags:
diff -Nur glib-2.12.4.orig/gthread-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/gthread-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/gthread-2.0-uninstalled.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gthread-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -2,5 +2,5 @@
 Description: Thread support for GLib, Not Installed
 Requires: glib-2.0-uninstalled
 Version: @VERSION@
-Libs: ${pc_top_builddir}/${pcfiledir}/gthread/libgthread-2.0.la @G_THREAD_LIBS@
+Libs: ${pc_top_builddir}/${pcfiledir}/gthread/libgthread-2.0.la @G_THREAD_LIBS@ @LIBRT@
 Cflags: -I${pc_top_builddir}/${pcfiledir}/@srcdir@ @G_THREAD_CFLAGS@
diff -Nur glib-2.12.4.orig/gthread-2.0.pc.in glib-2.12.4.pkgconfig-fix/gthread-2.0.pc.in
--- glib-2.12.4.orig/gthread-2.0.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/gthread-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -7,5 +7,5 @@
 Description: Thread support for GLib
 Requires: glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} -lgthread-2.0 @G_THREAD_LIBS@
+Libs: -L${libdir} -lgthread-2.0 @G_THREAD_LIBS@ @LIBRT@
 Cflags: @G_THREAD_CFLAGS@
