--- glib-2.14.2/configure.in.orig	2007-10-16 07:44:52.000000000 +0200
+++ glib-2.14.2/configure.in	2007-10-24 17:59:45.448866287 +0200
@@ -2198,8 +2198,8 @@
                       libpcre >= $PCRE_REQUIRED_VERSION)
     AC_CACHE_CHECK([for Unicode support in PCRE],glib_cv_pcre_has_unicode,[
                    glib_save_CFLAGS="$CFLAGS"
-                   glib_save_LDFLAGS="$LDFLAGS"
-                   CFLAGS="$CFLAGS $PCRE_CFLAGS" LDFLAGS="$LDFLAGS $PCRE_LIBS"
+                   glib_save_LIBS="$LIBS"
+                   CFLAGS="$CFLAGS $PCRE_CFLAGS" LIBS="$LIBS $PCRE_LIBS"
                    AC_TRY_RUN([#include <pcre.h>
                                int main () {
                                  int support;
@@ -2215,7 +2215,7 @@
                    glib_cv_pcre_has_unicode=no,
                    glib_cv_pcre_has_unicode=yes)
                    CFLAGS="$glib_save_CFLAGS"
-                   LDFLAGS="$glib_save_LDFLAGS"
+                   LIBS="$glib_save_LIBS"
 	])
     if test "$glib_cv_pcre_has_unicode" = "no"; then
       AC_MSG_ERROR([*** The system-supplied PCRE does not support Unicode properties or UTF-8.])
