--- glib-2.16.1/configure.in.orig	2008-03-30 21:42:29.515297272 +0200
+++ glib-2.16.1/configure.in	2008-03-30 21:45:58.500934165 +0200
@@ -1478,7 +1478,7 @@
 msg_selinux=no
 SELINUX_LIBS=
 if test "x$enable_selinux" != "xno"; then
-
+ TMP_LIBS=$LIBS
  AC_CHECK_LIB(selinux, is_selinux_enabled,
    [AC_CHECK_HEADERS(selinux/selinux.h,
      [AC_SEARCH_LIBS(lgetfilecon_raw, selinux, 
@@ -1487,6 +1487,7 @@
         msg_selinux=yes])
      ])
    ])
+ LIBS=$TMP_LIBS
 fi
 AC_SUBST(SELINUX_LIBS)
 
