--- glib-2.28.0/glib-2.0.pc.in~	2009-04-01 01:04:20.000000000 +0200
+++ glib-2.28.0/glib-2.0.pc.in	2011-02-11 08:47:51.847033632 +0100
@@ -10,6 +10,6 @@
 Name: GLib
 Description: C Utility Library
 Version: @VERSION@
 Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
-Libs.private: @ICONV_LIBS@
+Libs.private: @ICONV_LIBS@ -lrt
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
