diff -Nur glib-2.12.4.orig/configure.in glib-2.12.4.pkgconfig-fix/configure.in
--- glib-2.12.4.orig/configure.in	2006-10-01 05:51:15.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/configure.in	2006-11-05 20:44:19.000000000 +0000
@@ -863,8 +863,10 @@
   AC_CHECK_LIB(rt, clock_gettime, [
     AC_DEFINE(HAVE_CLOCK_GETTIME, 1)
     LIBS="$LIBS -lrt"
+    LIBRT="-lrt"
   ])
 ])
+AC_SUBST(LIBRT)
 
 AC_CACHE_CHECK(for monotonic clocks,
     glib_cv_monotonic_clock,AC_TRY_RUN([
diff -Nur glib-2.12.4.orig/glib-2.0-uninstalled.pc.in glib-2.12.4.pkgconfig-fix/glib-2.0-uninstalled.pc.in
--- glib-2.12.4.orig/glib-2.0-uninstalled.pc.in	2002-03-05 05:17:38.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/glib-2.0-uninstalled.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -5,5 +5,5 @@
 Name: GLib Uninstalled
 Description: C Utility Library, Not Installed
 Version: @VERSION@
-Libs: ${pc_top_builddir}/${pcfiledir}/glib/libglib-2.0.la @ICONV_LIBS@ @INTLLIBS@
+Libs: ${pc_top_builddir}/${pcfiledir}/glib/libglib-2.0.la @ICONV_LIBS@ @INTLLIBS@ @LIBRT@
 Cflags: -I${pc_top_builddir}/${pcfiledir} -I${pc_top_builddir}/${pcfiledir}/@srcdir@ -I${pc_top_builddir}/${pcfiledir}/@srcdir@/glib
diff -Nur glib-2.12.4.orig/glib-2.0.pc.in glib-2.12.4.pkgconfig-fix/glib-2.0.pc.in
--- glib-2.12.4.orig/glib-2.0.pc.in	2003-07-30 00:01:39.000000000 +0000
+++ glib-2.12.4.pkgconfig-fix/glib-2.0.pc.in	2006-11-05 20:46:27.000000000 +0000
@@ -10,6 +10,6 @@
 Name: GLib
 Description: C Utility Library
 Version: @VERSION@
-Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@
+Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@ @LIBRT@
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include 
 
