--- glib-2.69.1/docs/reference/gio/meson.build~	2021-08-04 17:34:39.000000000 +0200
+++ glib-2.69.1/docs/reference/gio/meson.build	2021-08-16 22:21:04.114468585 +0200
@@ -223,7 +223,7 @@
       '--extra-dir=' + join_paths('gio', '..', 'gobject', 'html'),
     ],
     install: true,
-    check: true,
+    check: false,
   )
 endif
 
--- glib-2.69.1/docs/reference/glib/meson.build~	2021-08-04 17:34:39.000000000 +0200
+++ glib-2.69.1/docs/reference/glib/meson.build	2021-08-16 22:21:41.744649339 +0200
@@ -88,7 +88,7 @@
                  '--html-dir=' + docpath,
                ],
                install: true,
-               check: true)
+               check: false)
 endif
 
 if get_option('man')
--- glib-2.69.1/docs/reference/gobject/meson.build~	2021-08-04 17:34:39.000000000 +0200
+++ glib-2.69.1/docs/reference/gobject/meson.build	2021-08-16 22:22:00.861422040 +0200
@@ -51,7 +51,7 @@
       '--extra-dir=' + join_paths('gobject', '..', 'glib', 'html'),
     ],
     install: true,
-    check: true,
+    check: false,
   )
 endif
 
